static/app/components/carousel.spec.tsx

```diff
@@ -1,79 +1,140 @@
-import {render, screen} from 'sentry-test/reactTestingLibrary';
+import {act, render, screen, userEvent} from 'sentry-test/reactTestingLibrary';
 
 import Carousel from 'sentry/components/carousel';
-import Placeholder from 'sentry/components/placeholder';
-
-export function setIntersectionObserver(
-  entries: {isIntersecting: boolean; target: {id: string}}[]
-) {
-  (() => {
-    return (global.IntersectionObserver = class IntersectionObserver {
-      [x: string]: any;
-      constructor(cb: any) {
-        this.cb = cb;
-      }
-      observe() {
-        this.cb(entries);
-      }
-      unobserve() {}
-      disconnect() {}
-    } as any);
-  })();
-}
 
 describe('Carousel', function () {
-  beforeEach(() => {});
-  it('hides arrows if content does not overflow in x', function () {
-    setIntersectionObserver([
-      {target: {id: 'left-anchor'}, isIntersecting: true},
-      {target: {id: 'right-anchor'}, isIntersecting: true},
-    ]);
+  let intersectionOnbserverCb: (entries: Partial<IntersectionObserverEntry>[]) => void =
+    jest.fn();
+
+  window.IntersectionObserver = class IntersectionObserver {
+    root = null;
+    rootMargin = '';
+    thresholds = [];
+    takeRecords = jest.fn();
 
+    constructor(cb: IntersectionObserverCallback) {
+      // @ts-expect-error The callback wants just way too much stuff for our simple mock
+      intersectionOnbserverCb = cb;
+    }
+    observe() {}
+    unobserve() {}
+    disconnect() {}
+  };
```

Comment:
Do we still need this if it's in `tests/js/setup`?

Reply:
yeah since I need to override the callback